src/main/java/edu/kit/hci/soli/controller/BookingsController.java

```diff
@@ -19,7 +19,10 @@ public class BookingsController {
     private VisitsRepository visitsRepository;
 
     @GetMapping("/bookings")
-    public String bookings(Model model, HttpServletResponse response, Principal principal) {
+    public String bookings(Model model, HttpServletResponse response, Principal principal, @AuthenticationPrincipal OidcUser oidcUser) {
+
+        String username = oidcUser.getUserInfo().getFullName();
+        model.addAttribute("model", new LoginStateModel(username, visitsRepository.getVisits()));
```

Comment:
oidcUser may be null. Also, the LoginStateModel is injected automatically

Reply:
This was a mistake i didnt mean to push this :))